src/Control/Category/Free.hs

```diff
@@ -212,28 +305,158 @@ between any two `CFree` functors.
+instance x ~ y => Semigroup (EndoL p x y) where (<>) = (>>>)
+instance x ~ y => Monoid (EndoL p x y) where mempty = id
 instance Category (EndoL p) where
   id = EndoL id
   EndoL f1 . EndoL f2 = EndoL (f1 . f2)
 
 {- | Turn a `Monoid` into a `Category`,
 used in the default definition of `ctoMonoid`.-}
-newtype MCat m x y = MCat {getMCat :: m} deriving (Eq, Ord, Show)
-instance Monoid m => Category (MCat m) where
-  id = MCat mempty
-  MCat g . MCat f = MCat (f <> g)
+newtype KQ m x y = KQ {getKQ :: m} deriving (Eq, Ord, Show)
+instance (Semigroup m, x ~ y) => Semigroup (KQ m x y) where
+  KQ f <> KQ g = KQ (f <> g)
+instance (Monoid m, x ~ y) => Monoid (KQ m x y) where mempty = id
+instance Monoid m => Category (KQ m) where
+  id = KQ mempty
+  KQ g . KQ f = KQ (f <> g)
 
 {- | Turn an `Applicative` over a `Category` into a `Category`,
 used in the default definition of `ctraverse_`.-}
-newtype ApCat m c x y = ApCat {getApCat :: m (c x y)} deriving (Eq, Ord, Show)
-instance (Applicative m, Category c) => Category (ApCat m c) where
-  id = ApCat (pure id)
-  ApCat g . ApCat f = ApCat ((.) <$> g <*> f)
+newtype ApQ m c x y = ApQ {getApQ :: m (c x y)} deriving (Eq, Ord, Show)
+instance (Applicative m, Category c, x ~ y)
+  => Semigroup (ApQ m c x y) where (<>) = (>>>)
+instance (Applicative m, Category c, x ~ y)
+  => Monoid (ApQ m c x y) where mempty = id
+instance (Applicative m, Category c) => Category (ApQ m c) where
+  id = ApQ (pure id)
+  ApQ g . ApQ f = ApQ ((.) <$> g <*> f)
+instance Functor t => CFunctor (ApQ t) where
+  cmap f (ApQ t) = ApQ (f <$> t)
+instance Applicative t => CPointed (ApQ t) where
+  csingleton = ApQ . pure
+instance Functor t => CStrong (ApQ t) where
+  csecond (ProductQ p (ApQ t)) = ApQ (ProductQ p <$> t)
+instance Applicative t => CApplicative (ApQ t) where
+  czip f (ApQ tp) (ApQ tq) = ApQ (f <$> tp <*> tq)
+instance Monad t => CMonad (ApQ t) where
+  cbind f (ApQ t) = ApQ $ do
+    p <- t
+    getApQ $ f p
+
+{- | Reverse all the arrows in a quiver.-}
+newtype OpQ c x y = OpQ {getOpQ :: c y x} deriving (Eq, Ord, Show)
+instance (Category c, x ~ y) => Semigroup (OpQ c x y) where (<>) = (>>>)
+instance (Category c, x ~ y) => Monoid (OpQ c x y) where mempty = id
+instance Category c => Category (OpQ c) where
+  id = OpQ id
+  OpQ g . OpQ f = OpQ (f . g)
+instance CFunctor OpQ where cmap f = OpQ . f . getOpQ
+
+{- | Turn all arrows in a quiver into bidirectional edges.-}
+data IsoQ c x y = IsoQ
+  { up :: c x y
+  , down :: c y x
+  } deriving (Eq, Ord, Show)
+instance (Category c, x ~ y) => Semigroup (IsoQ c x y) where (<>) = (>>>)
+instance (Category c, x ~ y) => Monoid (IsoQ c x y) where mempty = id
+instance Category c => Category (IsoQ c) where
+  id = IsoQ id id
+  (IsoQ yz zy) . (IsoQ xy yx) = IsoQ (yz . xy) (yx . zy)
+instance CFunctor IsoQ where
+  cmap f (IsoQ u d) = IsoQ (f u) (f d)
+
+{- | The identity functor on quivers. -}
+newtype IQ c x y = IQ {getIQ :: c x y} deriving (Eq, Ord, Show)
```

Comment:
`newtype Test x y = Test { runTest :: (y -> Bool) -> (x -> Bool) }`

`Test t . Test t' = Test $ \ctest -> (t . t')`

Heh, I think you get the joke

Reply:
The exercise of making `Test` a category made me want to make a `Generic` deriving mechanism for `Category`...